pyGHDL/dom/DesignUnit.py

```diff
@@ -61,54 +61,39 @@
     GetDeclaredItemsFromChainedNodes,
 )
 from pyGHDL.dom.Symbol import EntitySymbol
-from pyGHDL.dom.Common import GHDLMixin
 
 
 __all__ = []
 
 
 @export
-class Entity(VHDLModel_Entity, GHDLMixin):
+class Entity(VHDLModel_Entity):
     @classmethod
     def parse(cls, entityNode: Iir):
         name = GetNameOfNode(entityNode)
-        entity = cls(name)
-
-        for generic in GetGenericsFromChainedNodes(nodes.Get_Generic_Chain(entityNode)):
-            entity.GenericItems.append(generic)
-
-        for port in GetPortsFromChainedNodes(nodes.Get_Port_Chain(entityNode)):
-            entity.PortItems.append(port)
-
-        for item in GetDeclaredItemsFromChainedNodes(
+        generics = GetGenericsFromChainedNodes(nodes.Get_Generic_Chain(entityNode))
+        ports = GetPortsFromChainedNodes(nodes.Get_Port_Chain(entityNode))
+        declaredItems = GetDeclaredItemsFromChainedNodes(
             nodes.Get_Declaration_Chain(entityNode), "entity", name
-        ):
-            entity.DeclaredItems.append(item)
+        )
+        bodyItems = []
 
-        return entity
+        return cls(name, generics, ports, declaredItems, bodyItems)
```

Comment:
In cases such as this one, what do you think about the following style?

```py
        return cls(
            GetNameOfNode(entityNode),
            GetGenericsFromChainedNodes(nodes.Get_Generic_Chain(entityNode)),
            GetPortsFromChainedNodes(nodes.Get_Port_Chain(entityNode)),
            declaredItems,
            []
        )
````

I do understand the value of intermediate variables when those are used more than once, or when the "generation" of them is complex. However, most of these are two chained calls with a single input and a single output.

Reply:
Moreover, does it make sense to put the call to `Get_*_Chain` into the `Get*FromChainedNodes`, and rename it to `Get*FromEntityNode`?

Reply:
Maybe later when we are through all translations.